src/contracts/core/AVSDirectory.sol

```diff
@@ -249,114 +265,192 @@ contract AVSDirectory is
+                if (currentMagnitude == 0) {
+                    continue;
+                }
+
+                /// TODO: add wrapping library for rounding up for slashing accounting
+                slashedMagnitude = uint64(uint256(bipsToSlash) * uint256(currentMagnitude) / BIPS_FACTOR);
+
+                _magnitudeUpdate[operator][strategies[i]][msg.sender][operatorSetId].decrementAtAndFutureCheckpoints({
+                    key: uint32(block.timestamp),
+                    decrementValue: slashedMagnitude
+                });
+            }
+
+            // 2. if there are any pending deallocations then need to update and decrement if they fall within slashable window
+            // loop backwards through _queuedDeallocationIndices, each element contains an array index to
+            // corresponding deallocation to access in pendingFreeMagnitude
+            // if completable, then break
+            //      (since ordered by completableTimestamps, older deallocations will also be completable and outside slashable window)
+            // if NOT completable, then slash
+            {
+                uint256 queuedDeallocationIndicesLen =
+                    _queuedDeallocationIndices[operator][strategies[i]][msg.sender][operatorSetId].length;
+                for (uint256 j = queuedDeallocationIndicesLen; j > 0; --j) {
+                    // index of pendingFreeMagnitude/deallocation to check for slashing
+                    uint256 index =
+                        _queuedDeallocationIndices[operator][strategies[i]][msg.sender][operatorSetId][j - 1];
+                    PendingFreeMagnitude storage pendingFreeMagnitude =
+                        _pendingFreeMagnitude[operator][strategies[i]][index];
+
+                    // Reached pendingFreeMagnitude/deallocation that is completable and not within slashability window,
+                    // therefore older deallocations will also be completable. Since this is ordered by completableTimestamps break loop now
+                    if (pendingFreeMagnitude.completableTimestamp >= uint32(block.timestamp)) {
+                        break;
+                    }
+
+                    // pending deallocation is still within slashable window, slash magnitudeDiff and add to slashedMagnitude
+                    uint64 slashedAmount =
+                        uint64(uint256(bipsToSlash) * uint256(pendingFreeMagnitude.magnitudeDiff) / BIPS_FACTOR);
+                    pendingFreeMagnitude.magnitudeDiff -= slashedAmount;
+                    slashedMagnitude += slashedAmount;
+                }
+            }
+
+            // 3. update totalMagnitude, get total magnitude and subtract slashedMagnitude
+            _totalMagnitudeUpdate[operator][strategies[i]].push({
+                key: uint32(block.timestamp),
+                value: _getLatestTotalMagnitude(operator, strategies[i]) - slashedMagnitude
```

Comment:
Yup, since a deallocation immediately decrements current amount and places the decrement in a pendingFreeMagnitude.
When an operatorSet tries to slash, you need to slash from current and pendingFreeMagnitude amount to slash the same amount.